ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/operators/KeyedTableLookupOperator.java

```diff
@@ -90,19 +90,45 @@ public Object next() {
         if (!nextLocation.getKeys().isPresent()) {
           throw new IllegalStateException("Table lookup queries should be done with keys");
         }
-        keyIterator = nextLocation.getKeys().get().stream().map(KsqlKey::getKey).iterator();
+        keyIterator = nextLocation.getKeys().get().stream().iterator();
       }
       nextKey = keyIterator.next();
-      resultIterator = mat.nonWindowed()
-          .get(nextKey, nextLocation.getPartition())
-          .map(ImmutableList::of)
-          .orElse(ImmutableList.of()).iterator();
+      resultIterator = getMatIterator(nextKey);
     }
 
     returnedRows++;
     return resultIterator.next();
   }
 
+  private Iterator<Row> getMatIterator(final KsqlKey ksqlKey) {
+    if (!(nextKey instanceof KeyConstraint)) {
+      throw new IllegalStateException(String.format("Keyed lookup queries should be done with "
+        + "key constraints: %s", ksqlKey.toString()));
+    }
+    final KeyConstraint keyConstraintKey = (KeyConstraint) ksqlKey;
+    if (keyConstraintKey.getOperator() == ConstraintOperator.EQUAL) {
+      return mat.nonWindowed()
+        .get(ksqlKey.getKey(), nextLocation.getPartition())
+        .map(ImmutableList::of)
+        .orElse(ImmutableList.of()).iterator();
+    } else if (keyConstraintKey.getOperator() == ConstraintOperator.GREATER_THAN
```

Comment:
Yup, exactly, there is currently no other option. In the future we might want to consider change the range interface of the state stores to be more expressive so that we don't need extra filtering to trim the borders.